lib/dor/models/identifiable.rb

```diff
@@ -224,6 +224,19 @@ def remove_druid_prefix(druid=id)
       result.nil? ? druid : result[0]  # if no matches, return the string passed in, otherwise return the match 
     end
 
+    # Override ActiveFedora::Core#adapt_to_cmodel (used with associations, among other places) to
+    # preferentially use the objectType asserted in the identityMetadata.
+    def adapt_to_cmodel
+      object_type = identityMetadata.objectType.first
+      object_class = Dor.registered_classes[object_type]
+
+      if object_class
```

Comment:
I think it's just `nil`. We were guarding against it before too: https://github.com/sul-dlss/dor-services/pull/248/files#diff-0d90e7b0ff1134056873ce5aa4b3e0f1L26